src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -0,0 +1,164 @@
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import com.revrobotics.CANEncoder;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Game;
+import frc.robot.MotorController;
+import frc.robot.Robot;
+
+public class ShooterSubsystem extends SubsystemBase {
+  /**
+   * Creates a new ShooterSubsystem.
+   */
+  private final MotorController mShooterMotor;
+  private final DoubleSolenoid mHoodDoubleSolenoid;
+  private final CANPIDController mShooterPID;
+  private double mShooterVelocitySetpoint = 0.0;
+  
+  private double mP = 0.0;
+  private double mI = 0.0;
+  private double mD = 0.0;
+
+  private double mGreenShootingVelocity = 0.0;
+  private double mYellowShootingVelocity = 0.0;
+  private double mBlueShootingVelocity = 0.0;
+  private double mRedShootingVelocity = 0.0;
+
+  public ShooterSubsystem() {
+    mShooterMotor = new MotorController("Shooter Motor", Constants.kShooterMotorPort, Constants.kShooterMotorCurrentLimit);
+    mHoodDoubleSolenoid = new DoubleSolenoid(Constants.kHoodDoubleSolenoidForwardChannel, Constants.kHoodDoubleSolenoidReverseChannel);
+    mShooterPID = mShooterMotor.getSparkMax().getPIDController();
+    mShooterPID.setP(mP);
+    mShooterPID.setI(mI);
+    mShooterPID.setD(mD);
+    SmartDashboard.putNumber("Shooter P Value", mP);
+    SmartDashboard.putNumber("Shooter I Value", mI);
+    SmartDashboard.putNumber("Shooter D Value", mD);
+  }
+
+  // Value.kReverse is when the hood is extended
+  public void setHoodExtended(boolean isExtended) {
+    if (isExtended)
+      mHoodDoubleSolenoid.set(Value.kReverse);
+    else  
+      mHoodDoubleSolenoid.set(Value.kForward);
+  }
+  
+  public boolean getHoodExtendRequired() {
+    return getDistanceFromGoal() > Constants.kHoodExtendRequiredDistance;
+  }
+
+  public void setVelocity(double velocity) {
+    mShooterVelocitySetpoint = velocity;
+    mShooterPID.setReference(velocity, ControlType.kVelocity);
+  }
+
+  public boolean getMotorVelocityWithinRange() {
+    // Compare if the motor velocity is within Constants.kShooterVelocityRange range
+    return (Math.abs(mShooterVelocitySetpoint - mShooterMotor.getEncoder().getVelocity()) < mShooterVelocitySetpoint * Constants.kShooterVelocityRange);
+  }
+
+  public boolean getRobotDistanceWithinRange() {
+    return getDistanceFromGoal() < getDistanceLimit();
+  }
+
+  public double getDistanceFromGoal() {
+    //Todo: get distance from limelight
+    return SmartDashboard.getNumber("Distance from Goal", 0.0);
+  }
+
+  public double getAngleFromGoal() {
+    //Todo: get angle from limelight
+    return SmartDashboard.getNumber("Angle from Goal", 0.0);
+  }
+
+  public boolean getAngleAligned() {
+    return getAngleFromGoal() == 0;
+  }
+
+  public boolean getReadyToShoot() {
```

Comment:
Consider using 'is' instead of 'get' for when doing boolean check functions to make it clearer what the functions do. 
e.g. isReadyToShoot() makes it clear we are just checking that we are ready to shoot, where getReadyToShoot() could be interpreted as causing the Robot to take an action to get ready to shoot.